### ISSUE TITLE
2.0.0 trash call is broken

### ISSUE THREAD

**Issue by MoOx** [MoOx/npmpub#4]
/Users/MoOx/Sync/Development/statinamic/node_modules/.bin/npmpublish: line 23: /Users/MoOx/Sync/Development/statinamic/node_modules/npmpub/node_modules/.bin/trash: No such file or directory

**Comment by MoOx** [comment]
@jeddy3 are you using npm@2 or npm@3?

**Comment by MoOx** [comment]
Because with npm@3, trash bin is at the same level as npmpublish bin, so the "fix" that we introduced is only relevant for an package installed with npm@2, used with npm@3 (eg: `nvm install 5 --reinstall-packages-from=4`)

**Comment by MoOx** [comment]
@jeddy3
- Did you get issue with 1.x with npm@2
- Did you get issue with 1.x with npm@3
- Is npmpub@2 working for you with npm@2
- Is npmpub@2 working for you with npm@3

For me, I need to use 1.x version with npm@3 (like I was with npm@2).

**Comment by jeddy3** [comment]
I'm using npm@3.

I'm afraid I never tried 1.x as I suggested the update to trash after getting a deprecated message during the install. I then pulled that version into stylelint.

I did get an authentication error when I tried to use npmpub@2, but I think that's just my tokens being messed up. I was going to wait for another release of stylelint or config-suit to try and resolve that. I don't  recall getting the `No such file or directory` error that you mention above.

@ntwb You used npmpub@2 to publish config-wordpress, right? What version of npm are using and did you encounter any errors?

**Comment by MoOx** [comment]
(note: the error is "just" a one line message and can easily be not seen)

**Comment by jeddy3** [comment]
There's a good chance I missed it then as the authentication error drew my attention.

**Comment by jeddy3** [comment]
FYI, I did see the error when releasing stylelint 4.0.0 just now :(

**Comment by MoOx** [comment]
Ok, so I will revert this part of the change as the mix of 1.x and 2.x is working (./node_modules/.bin/trash using trash-cli).

**Comment by ntwb** [comment]
I'm using npm v3.5.3 alongside node v5.4.1, I didn't see any error or errors.

If there was an error I think I would have seen it as as I was watching very closely for any issues since this was the first time it had been run on that repo or locally myself.

**Comment by MoOx** [comment]
Anyway, I am rewriting this as a node script, that will be simpler to debug/trace.

**Comment by jeddy3** [comment]
> Anyway, I am rewriting this as a node script,

:+1: 

P.S. @MoOx Thanks for this module btw. It made that release I just did so so simple :)

**Comment by MoOx** [comment]
@jeddy3 what do you think of this

```
npmpublish [options]

  --help          Just what you are reading.
  --verbose       Get some informations.
  --debug         Get all informations about process.
  --skip-status   Skip git status check (⚠︎ you might release unversionned stuff).
  --skip-fetch    Skip git fetch to compare remote (⚠︎ you might not be able to push).
  --skip-compare  Skip git comparison with origin (⚠︎ you might not be able to push).
  --skip-cleanup  Skip node_modules cleanup (⚠︎ you might miss some dependencies changes).
  --dry           No publish, just check that tests are ok.
  --no-release    No GitHub release from changelog.
```

v3 will be written in JS and will include and use github-release-from-changelog by default.

Any thoughts?

**Comment by jeddy3** [comment]
Looks good to me! :)

The `--dry` flag is a nice touch. As is including `github-release-from-changelog` by default, with a flag to turn it off. Those two things alone will make this tool even easier in my case.

Thanks!

**Comment by MoOx** [comment]
trash cli is not used anymore in v3 (but trash node api is).

Closed by d8e09b77996c829dfa1e4282072f7976acdcbb9d

**Comment by ntwb** [comment]
Agreed, `--dry` is nice, I knew when I ran this the first time _something was going to happen, good, or bad, something was going to happen_ :smirk:
